src/context.ts

```diff
@@ -9,8 +10,13 @@ const yaml = require('js-yaml')
  * @property {payload} payload - The webhook event payload
  * @property {logger} log - A logger
  */
-class Context {
-  constructor (event, github, log) {
+export class Context {
+  id?: string
+  github: GitHubApi
+  log: LoggerWithTarget
+  payload!: WebhookPayloadWithRepository
```

Comment:
I think that it should probably be something like `WebhookPayloadWithRepository<T>` so that we can pass the type of Github payload that it contains

Reply:
:+1: for adding generics to the payload. I am not sure how that would work with `robot.on(eventName, handler)` but I think you can specify something like:

```typescript
type PushPayload = { /* ... */ }
type IssuesPayload = { /* ... */ }

interface IRobot {
  on(eventName: 'push',   callback: ({payload: PushPayload})   => void)
  on(eventName: 'issues', callback: ({payload: IssuesPayload}) => void)
}
```